nova_vm/src/ecmascript/builtins/array.rs

```diff
@@ -178,33 +211,118 @@ impl InternalMethods for Array {
 
     fn get_own_property(
         self,
-        _agent: &mut Agent,
-        _property_key: crate::ecmascript::types::PropertyKey,
-    ) -> JsResult<Option<crate::ecmascript::types::PropertyDescriptor>> {
-        todo!()
+        agent: &mut Agent,
+        property_key: PropertyKey,
+    ) -> JsResult<Option<PropertyDescriptor>> {
+        if let PropertyKey::Integer(index) = property_key {
+            let elements = agent.heap.get(*self).elements;
+            let elements = agent.heap.elements.get(elements);
+            if let Some(value) = elements.get(index.into_i64() as usize) {
+                return Ok(value.map(|value| PropertyDescriptor {
+                    value: Some(value),
+                    ..Default::default()
+                }));
+            }
+            return Ok(None);
+        }
+        let length_key = PropertyKey::from_str(&mut agent.heap, "length");
+        let array_data = agent.heap.get(*self);
+        if let Some(object_index) = array_data.object_index {
+            Object::Object(object_index).get_own_property(agent, property_key)
+        } else if property_key == length_key {
+            Ok(Some(PropertyDescriptor {
+                value: Some(array_data.elements.len().into()),
+                ..Default::default()
+            }))
+        } else {
+            Ok(None)
+        }
     }
 
     fn define_own_property(
         self,
-        _agent: &mut Agent,
-        _property_key: crate::ecmascript::types::PropertyKey,
-        _property_descriptor: crate::ecmascript::types::PropertyDescriptor,
+        agent: &mut Agent,
+        property_key: PropertyKey,
+        property_descriptor: PropertyDescriptor,
     ) -> JsResult<bool> {
-        todo!()
+        if property_key == PropertyKey::SmallString(SmallString::try_from("length").unwrap()) {
+            array_set_length(agent, self, property_descriptor)
+        } else if let PropertyKey::Integer(index) = property_key {
+            let index = index.into_i64();
+            if !(0 <= index && index <= (i64::pow(2, 32) - 2)) {
```

Comment:
Just eye candy. Also, did you mean to exclude the maximum `u32` value? This range can be modified to `..=` to include all valid `u32` values.

```suggestion
            if !(0..u32::MAX as i64).contains(&index) {
```

Reply:
Yeah I think I did mean that :) Just translating spec text without thinking about it properly